### PR TITLE
export targets in a addition to include directories / libraries

### PR DESCRIPTION
**Pull request by dirk-thomas**
Allowing modern CMake usage.

### CODE REVIEW

**Comment by dirk-thomas** [comment]
CI builds using only FastRTPS:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/10293/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/5827/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/8373/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/10190/)

File: rosidl_runtime_c/include/rosidl_runtime_c/primitives_sequence_functions.h [hunk 1]
```diff
@@ -26,6 +26,21 @@ extern "C"
...
+/// Allocate the memory for the sequence.
+/* Calling the function with an already allocated sequence will leak the
+  * previously allocated memory.
+  *
+  * \param sequence a pointer to a sequence struct
+  * \param size the number of items to allocate in the sequence, both sequence
+  *   fields `size` and `capacity` are set to this parameter
+  * \return true if successful, false if the passed sequence pointer is null
+  *   or the memory allocation failed
+  */
```
Comment (hidmic) [inline rosidl_runtime_c/include/rosidl_runtime_c/primitives_sequence_functions.h:38]: @dirk-thomas nit: bad indent?

File: rosidl_runtime_c/include/rosidl_runtime_c/primitives_sequence_functions.h [hunk 2]
```diff
@@ -26,6 +26,21 @@ extern "C"
...
+/// Deallocate the memory of the sequence.
+/* Calling the function with an already deallocated sequence is a no-op.
+  *
+  * \param sequence a pointer to a sequence struct
+  */
```
Comment (hidmic) [inline rosidl_runtime_c/include/rosidl_runtime_c/primitives_sequence_functions.h:43]: @dirk-thomas same as above?

**Comment by dirk-thomas** [comment]
> why mixing CMake and C/C++ documentation updates in the same commit?

Good catch. I accidentally used `-a` with other changes in the working copy. I removed the documentation related changes.

**Comment by dirk-thomas** [comment]
Related to ros2/ros2#904.

**Pull request merged**